plugin/core/open.py

```diff
@@ -2,17 +2,69 @@
 from .promise import Promise
 from .promise import ResolveFunc
 from .protocol import DocumentUri
+from .protocol import UINT_MAX
 from .protocol import Range
 from .protocol import RangeLsp
 from .typing import Dict, Tuple, Optional
 from .url import parse_uri
 from .views import range_to_region
+from urllib.parse import unquote, urlparse
 import os
+import re
 import sublime
 import subprocess
+import webbrowser
 
 
 opening_files = {}  # type: Dict[str, Tuple[Promise[Optional[sublime.View]], ResolveFunc[Optional[sublime.View]]]]
+FRAGMENT_PATTERN = re.compile(r'^L?(\d+)(?:,(\d+))?(?:-L?(\d+)(?:,(\d+))?)?')
+
+
+def open_file_uri(
+    window: sublime.Window, uri: DocumentUri, flags: int = 0, group: int = -1
+) -> Promise[Optional[sublime.View]]:
+
+    def parse_fragment(fragment: str) -> RangeLsp:
+        match = FRAGMENT_PATTERN.match(fragment)
+        if match:
+            # assume that line and column numbers in the fragment are 1-based
+            start_line, start_column, end_line, end_column = [max(1, int(g)) if g else None for g in match.groups()]
+            if start_line is not None:
+                if end_line is not None:
+                    if start_column is not None and end_column is not None:
+                        return {
+                            "start": {"line": start_line - 1, "character": start_column - 1},
+                            "end": {"line": end_line - 1, "character": end_column - 1}
+                        }
+                    else:
+                        return {
+                            "start": {"line": start_line - 1, "character": 0},
+                            "end": {"line": end_line - 1, "character": UINT_MAX}
+                        }
+                elif start_column is not None:
+                    return {
+                        "start": {"line": start_line - 1, "character": start_column - 1},
+                        "end": {"line": start_line - 1, "character": start_column - 1}
+                    }
+                else:
+                    return {
+                        "start": {"line": start_line - 1, "character": 0},
+                        "end": {"line": start_line - 1, "character": 0}
+                    }
+        return {"start": {"line": 0, "character": 0}, "end": {"line": 0, "character": 0}}
```

Comment:
to reduce the indent, how about something like this: 
```py
    def parse_fragment(fragment: str) -> RangeLsp:
        match = FRAGMENT_PATTERN.match(fragment)
        result = {"start": {"line": 0, "character": 0}, "end": {"line": 0, "character": 0}}  # type: RangeLsp
        if match:
            # assume that line and column numbers in the fragment are 1-based
            start_line, start_column, end_line, end_column = [max(1, int(g)) if g else None for g in match.groups()]
            if start_line:
                result['start']['line'] = start_line - 1
            if start_column:
                result['start']['character'] = start_column - 1
                result['end']['character'] = start_column - 1
            if end_line:
                result['end']['line'] = end_line - 1
            if end_column:
                result['end']['character'] = end_column - 1
        return result
```

Reply:
Added with a few adjustments. It requires 1-2 more if-checks than before and possibly several overrides of the result dict, but that shouldn't really matter and it appears visually more pleasant this way. Should be equivalent to the replaced code now.

Reply:
Doesn't the lines still have to be compared against `None`? I'm looking at this on mobile so maybe I'm missing something but line 0 would evaluate to false.

Reply:
> Doesn't the lines still have to be compared against None? I'm looking at this on mobile so maybe I'm missing something but line 0 would evaluate to false.

Initially I thought that as well, but 0 is already the default for all of the values, so it wouldn't make any difference to overwrite it with 0 again.

I think there are only two differences to the `is not None` check, that could theoretically occur:
* `end_line` is 0, but `start_line` is not -> I would assume that for valid ranges, "end" should come after "start", so this should not hopefully not happen (and if it happens, with the current code we would ignore the assumingly faulty "end" to fix it)
* the range spans several lines, and at the same time the `end_column` is set to 0. In other words, it should include the newline character. Yeah, I guess this might be a valid usecase, so I better adjust the if-check for that.